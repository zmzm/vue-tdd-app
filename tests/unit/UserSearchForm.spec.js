import { shallowMount } from '@vue/test-utils';

import UserSearchForm from '@/components/UserSearchForm';

describe('UserSearchForm', () => {
  const build = () => {
    const wrapper = shallowMount(UserSearchForm);

    return {
      wrapper,
      input: () => wrapper.find('input'),
      button: () => wrapper.find('button'),
    };
  };

  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders the component', () => {
    // arrange
    const { wrapper } = build();
    // assert
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders main components', () => {
    // arrange
    const { input, button } = build();
    // assert
    expect(input().exists()).toBe(true);
    expect(button().exists()).toBe(true);
  });
});
