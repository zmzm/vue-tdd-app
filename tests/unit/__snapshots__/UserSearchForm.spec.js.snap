// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSearchForm renders the component 1`] = `<form><input type="text"><button type="submit">Search</button></form>`;
